modules/gdscript/gdscript_parser.h

```diff
@@ -1560,7 +1561,7 @@ class GDScriptParser {
 #endif // TOOLS_ENABLED
 
 public:
-	Error parse(const String &p_source_code, const String &p_script_path, bool p_for_completion);
+	Error parse(const String &p_source_code, const String &p_script_path, bool p_for_completion, bool p_parse_body = true);
```

Comment:
Instead of having a short-circuit parameter, I would just create a new method for the class.

Reply:
Just to be sure I understand correctly, you would prefer a new method like `parse_header` that only takes the parameters `p_source_code` and `p_script_path`, and remove the parameter `p_parse_body` from `parse`?

I suggest renaming the current `parse` to `_parse` and making it private, then creating a new method `parse` that calls `_parse` with `p_parse_body` set to true.
Like that:
```cpp
private:
Error _parse(const String &p_source_code, const String &p_script_path, bool p_for_completion, bool p_parse_body = true);
public:
Error parse(const String &p_source_code, const String &p_script_path, bool p_for_completion);
Error parse_header(const String &p_source_code, const String &p_script_path);
```
Does it sounds a good idea? I don't want to duplicate the code from `parse` method.

Reply:
I don't see why adding an optional parameter in an unexposed method would be a problem. Unless it's used with `callable_mp()` somewhere.